### PR TITLE
set the correct size of aten tensor in presence of mkldnn padding

### LINKED ISSUE #86556: `optimize_for_inference` leads to wrong results for model with conv2d, max and clip
### 🐛 Describe the bug

Hi! I find that the following model gives wrong results after `optimize_for_inference`. I think there's some issue with the optimization passes.

```python
import torch
import torch.nn as nn

class MyModule(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv1 = nn.Conv2d(
            1, 2, kernel_size=(2,4), stride=2, padding=2, dilation=(2,1),
        )
        
    def forward(self, i0):
        x = self.conv1(i0)
        o0 = torch.max(x, i0)
        o1 = torch.clip(x, -1.5, 1.5)
        return o0, o1



inp = {
    'i0': torch.zeros((1,1,1,2), dtype=torch.float32),
}

mod = MyModule()

out = mod(**inp)
print(f'{out = }')

exported = torch.jit.trace(mod, list(inp.values()))
exported = torch.jit.optimize_for_inference(exported)

eout = exported(**inp) # <-- wrong results here
print(f'{eout = }')

eqs = []
for x, y in zip(out, eout):
    eqs.append(torch.allclose(x, y))
print(eqs)
assert all(eqs)
```

### Versions

```python
PyTorch version: 1.12.1+cu116
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.9.13 (main, Aug 25 2022, 23:26:10)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-48-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.6.124
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090

Nvidia driver version: 515.65.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.3
[pip3] torch==1.12.1+cu116
[pip3] torchaudio==0.12.1+cu116
[pip3] torchvision==0.13.1+cu116
[conda] numpy                     1.23.3                   pypi_0    pypi
[conda] torch                     1.12.1+cu116             pypi_0    pypi
[conda] torchaudio                0.12.1+cu116             pypi_0    pypi
[conda] torchvision               0.13.1+cu116             pypi_0    pypi
```

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

### PR DESCRIPTION
**Pull request by Krovatkin**
This fixes https://github.com/pytorch/pytorch/issues/86556

### CODE REVIEW

**Review by eellison — APPROVED** [review]
nice!!
